docs/adr/0005-feature-flags.md

```diff
@@ -0,0 +1,145 @@
+* CICI: simpler merges into the main branch
+* Environment-dependent configurations
+* Experimentations: A/B testing and different experiences for different users based on context (user, device, geolocation data etc)
+
+## Decision Drivers <!-- optional -->
+
+When applying feature flagging, we often have a choice between making that toggle on the frontend or the backend. Here are a few factors that we should keep in mind while making that decision
+
+### UI Performance
+By moving flagging decisions to the server side, we are reducing code bloat on the frontend, gaining performance that's immediately noticeable by the user. If we decide to move Monitor frontend to SPA in the future, Single-page applications are already making a API call to render the data needed for the UI. With the same payload, we can make a call to a feature-flag service, so one network call can contain all feature-flag configs with the server-side data.
+
+### Config Cache Lag 
+There are two general approaches:
+
+1. An app can proactively request flagging decisions based on runtime-context
+
+> This approach consumes a lot more bandwidth (every context changes = new feature flag decision being made by the backend). It's a lot more flexible and prevents cache lag
+2. An app can request general config and use client-side router
+
+> With this approach, we essentially cache the config. In the case of a kill switch, the frontend may not be able to react quickly to the changes from the backend.
+
+### Security
+One key consideration when deciding where to implement a feature flag is security. If a feature is highly sensitive or critical to the security of the application, it may be safer to implement the toggle on the backend. This can help ensure that the feature is properly secured and that user data is protected.
+
+### Sync Complexity
+Implementing feature flagging on the server side reduces the complexity of synchronizing feature toggle logic between the frontend and backend, as there is a single central location to manage the toggle settings. This simplifies development and ensures that the system runs smoothly, as dependencies flow in one direction.
+## Maintenance & Best Practices
+### Naming Flags
+Defining a pattern or naming convention is generally a good practice for keeping things organized. Engineers can quickly identify from the name what the feature is about and be able to recognize what areas of the application for which the feature is being used. I'll list a few useful examples of naming structures and demonstrate the practical benefits.
+
+A good naming **convention** I found is as follows:
+```
+section_featurepurpose_service
+```
+
+The feature name example that follows has three parts:
+1. section: we present the name of the section the feature is gating. 
+2. purpose: indicates what the feature does
+3. layer: where in the stack the feature is located
```

Comment:
Not sure if these align with the example above.
Should it be this?
1. `section`: &hellip;
2. `featurepurpose`: &hellip;
3. `service`: &hellip;

Or, should `section_featurepurpose_service` change? (if we do change it, "featurepurpose" feels a bit of a mouthful).

Reply:
Thanks Peter! Changed